v0/destinations/kafka/transform.js

```diff
@@ -1,36 +1,83 @@
+/* eslint-disable no-restricted-syntax */
+const groupBy = require("lodash/groupBy");
+const cloneDeep = require("lodash/cloneDeep");
+const { getDynamicMeta } = require("../../../adapters/utils/networkUtils");
 const { getIntegrationsObj } = require("../../util");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const ErrorBuilder = require("../../util/error");
 
 function batch(destEvents) {
   const respList = [];
-  const batchedRequest = [];
-  const metadata = [];
-  destEvents.forEach(event => {
-    metadata.push(event.metadata);
-    batchedRequest.push(event.message);
-  });
-  respList.push({
-    batchedRequest,
-    metadata,
-    destination: destEvents[0].destination
+
+  // Grouping the events by topic
+  const groupedEvents = groupBy(destEvents, event => {
+    return event.message.topic;
   });
 
+  // Creating a batched request for each topic
+  // we are grouping the events based on topics
+  // and creating a batched request for each topic
+  // example: input events = [{event1,topic1},{event2,topic1},{event3,topic2}]
+  // out from transformer:  {batchedRequest:[{event1},{event2}]}, {batchedRequest:[{event3}]} (2 multilexed responses)
+  for (const events of Object.values(groupedEvents)) {
+    const response = {};
+    response.batchedRequest = events.map(event => event.message);
+    response.metadata = events.map(event => event.metadata);
+    response.destination = events[0].destination;
+    respList.push(response);
+  }
+
   return respList;
 }
 
 function process(event) {
-  const integrationsObj = getIntegrationsObj(event.message, "kafka");
+  const { message, destination } = event;
+  const integrationsObj = getIntegrationsObj(message, "kafka");
   const { schemaId } = integrationsObj || {};
+  const topic = integrationsObj?.topic || destination.Config.topic;
```

Comment:
is destination.Config.topic going to be dynamic field like {{}} this?

Reply:
That would be handled before this call is made at versionedRouter